pkg/app/piped/planner/kubernetes/kubernetes.go

```diff
@@ -480,3 +492,42 @@ func determineVersion(manifests []provider.Manifest) (string, error) {
 
 	return b.String(), nil
 }
+
+// determineVersions decides artifact versions of an application.
+// It finds all container images that are being specified in the workload manifests then returns their names, version numbers, and urls.
+func determineVersions(manifests []provider.Manifest) ([]*model.ArtifactVersion, error) {
+	imageMap := map[string]struct{}{}
+	for _, m := range manifests {
+		// TODO: Determine container image version from other workload kinds such as StatefulSet, Pod, Daemon, CronJob...
+		if !m.Key.IsDeployment() {
+			continue
+		}
+		data, err := m.MarshalJSON()
+		if err != nil {
+			return nil, err
+		}
+		var d resource.Deployment
+		if err := json.Unmarshal(data, &d); err != nil {
+			return nil, err
+		}
+
+		containers := d.Spec.Template.Spec.Containers
+		// remove duplicate images on multiple manifests
```

Comment:
```suggestion
		// Remove duplicate images on multiple manifests.
```
As our convention, comments start with a capital character and finish with a dot.